client/src/components/Filters.vue

```diff
@@ -1,9 +1,41 @@
 <template>
     <div v-for="filter in filters" :key="filter.id" class="filter-container">
-        <div v-if="filter.type === 'empty' || filter.isSelectionOpen" class="type-selection">
+        <div v-if="filter.type === 'date' && (filter.value.length > 7 || filter.isSelectionOpen)">
```

Comment:
If you enter an invalid date and then click away, you lose the "add date" option forever.
![image](https://user-images.githubusercontent.com/646155/141117695-e8a4f224-bd6c-45dc-b110-f781025ef2fd.png)

Reply:
@micahh2 thank you for noting this! I'm not sure I changed anything significant to handle this case but i'm having difficulties replicating the behavior. I tried with letters and two digits only - seems to show the chip on my side. Maybe I haven't considered the edge case you tried. Do you remember what kind of invalid date you used?

Reply:
@AlbinaMuzafarova Sorry I didn't provide more detailed steps!
1) Open Options Menu & Click add 
2) Click "+ date"
2) Enter 123 and 123 for the two dates
4) Click away (or let edit timeout)

Oddly enough it sometimes shows up again once the page is reloaded, but only if you've first added another "empty" filter.

Reply:
I looks like on the latest version of your branch this behavior is fixed. :+1:

Reply:
thank you Micah, that helps a lot! 

I will keep this thread open until next feedback, although it seems to behave alright now.